src/_plugin.ts

```diff
@@ -66,6 +66,10 @@ export function wrapFetch(server: Server, fetchHandler: ServerHandler) {
       }
     }
 
+    if (resPromise && server.options.onError) {
```

Comment:
I think we should implement it on entry level not plugin + we need to also catch non async errors. For this, runtimes themselve have facility

Reply:
Updated!

Moved the error handling out of the _plugins helper.

Also created a sync version of the tests